http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala

```diff
@@ -245,8 +246,9 @@ class GKEInterpreter[F[_]: Parallel: ContextShift: Timer](
       dbClusterOpt <- kubernetesClusterQuery.getMinimalClusterById(dbNodepool.clusterId).transaction
       dbCluster <- F.fromOption(
         dbClusterOpt,
-        KubernetesClusterNotFoundException(
-          s"Cluster with id ${dbNodepool.clusterId} not found in database | trace id: ${ctx.traceId}"
+        BadRequestException(
```

Comment:
Isn't `KubernetesClusterNotFoundException` more accurate of an exception in this case? Could we update that sort of exception to handle the traceId the way a `BadRequestException` type does instead of switching the types here?

Reply:
it is...two reasons why switching to `BadRequestException`

1. `KubernetesClusterNotFoundException` is an exception defined in wb-libs, and it doesn't have `traceId` field.
2. I don't think it's super helpful to have a unique type for various "bad requests". The message itself should be unique so that we can easily identify where the error happens.

Reply:
The name `KubernetesClusterNotFoundException` in this context threw me off in the past because it sounds like the caller requests some info about a cluster (like in a `GET cluster` request) that doesn't exist whereas what's happening here is specific to our implementation (i.e. registering a record in Leo DB before doing anything else on the Google side). I don't think the name `BadRequestException` accurately describes the situation either. How about `CreateClusterException`?